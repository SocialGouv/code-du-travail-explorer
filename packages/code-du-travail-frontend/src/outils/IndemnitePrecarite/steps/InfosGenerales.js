import React from "react";
import { Alert } from "@cdt/ui";
import { Field } from "react-final-form";

import { TypeContrat } from "../components/TypeContrat";
import { QuestionLabel } from "../../common/stepStyles";
import { YesNoQuestion } from "../../IndemniteLicenciement/components/YesNoQuestion";

function StepInfosGenerales() {
  return (
    <>
      <TypeContrat name="contrat" />
      <Field name="contrat">
        {({ input }) => {
          switch (input.value) {
            case "cdd":
              return (
                <>
                  <QuestionLabel as="p">
                    Votre type de CDD fait-il partie de la liste suivante :
                  </QuestionLabel>
                  <ul>
                    <li>CDD saisonnier ou CDD d’usage</li>
                    <li>
                      Contrat unique d’insertion (CUI) ou Parcours emploi
                      compétences (PEC)
                    </li>
                    <li>
                      Contrat de professionnalisation ou Contrat d’apprentissage
                    </li>
                    <li>
                      CDD conclu avec un jeune (mineur ou majeur) pendant ses
                      vacances scolaires ou universitaires
                    </li>
                    <li>
                      Contrat pour lequel l’employeur s’est engagé à assurer un
                      complément de formation professionnelle au salarié
                    </li>
                    <li>CCD dans le cadre d’un congé de mobilité</li>
                  </ul>
                  <YesNoQuestion label="" name="typeCdd" />
                </>
              );
            case "ctt":
              return (
                <>
                  <YesNoQuestion
                    label="Avez vous un contrat de mission-formation"
                    name="cttFormation"
                  />
                  <Field name="cttFormation">
                    {({ input }) =>
                      input.value === false ? (
                        <Alert variant="info">
                          Attention : si vous avez un contrat de travail
                          temporaire (contrat d’intérim) saisonnier ou d’usage,
                          un accord d’entreprise ou d’établissement peut
                          dispenser votre entreprise de travail temporaire
                          (l’entreprise d’intérim) de vous verser une indemnité
                          de précarité.
                        </Alert>
                      ) : null
                    }
                  </Field>
                </>
              );
            default:
              return null;
          }
        }}
      </Field>
    </>
  );
}
StepInfosGenerales.validate = function(values) {
  const errors = {};
  if (values.typeCdd === true) {
    errors.typeCdd =
      "Votre type de contrat ne vous permet pas d’avoir droit à une prime de précarité. ";
  }
  if (values.cttFormation === true) {
    errors.cttFormation =
      " Votre type de contrat ne vous permet pas d’avoir droit à une prime de précarité.";
  }
  return errors;
};
export { StepInfosGenerales };
