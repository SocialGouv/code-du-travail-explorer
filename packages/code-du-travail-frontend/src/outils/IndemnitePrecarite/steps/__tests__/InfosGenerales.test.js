import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { StepInfosGenerales } from "../InfosGenerales";
import { Form } from "react-final-form";

function renderForm(data) {
  return render(
    <Form
      validate={StepInfosGenerales.validate}
      initialValues={{ ...data }}
      onSubmit={jest.fn()}
    >
      {({ form }) => <StepInfosGenerales form={form} />}
    </Form>
  );
}

describe("<StepInfosGenerales />", () => {
  it("should render", () => {
    const { container } = renderForm({});
    expect(container).toMatchSnapshot();
  });

  it("should render with CDD", () => {
    const { container } = renderForm({ contrat: "cdd" });
    expect(container).toMatchSnapshot();
  });

  it("should render with CTT", () => {
    const { container } = renderForm({ contrat: "ctt" });
    expect(container).toMatchSnapshot();
  });

  it("should display error if contrat cdd and belong to special cases", () => {
    const { getByLabelText, getByText } = renderForm({ contrat: "cdd" });
    //trigger validation
    const specialCase = getByLabelText(/oui/i);
    fireEvent.click(specialCase);
    // blur is need to force validation in react-testing-lib
    fireEvent.blur(specialCase);
    expect(getByText(/ne vous permet pas/i)).toBeTruthy();
  });

  it("should display and error if ctt and contrat is mission-formation", () => {
    const { getByLabelText, getByText } = renderForm({ contrat: "ctt" });
    const missionFormation = getByLabelText(/oui/i);
    fireEvent.click(missionFormation);
    // blur is need to force validation in react-testing-lib
    fireEvent.blur(missionFormation);
    expect(getByText(/ne vous permet pas/i)).toBeTruthy();
  });
  it("should display and info alert if ctt and contrat is not mission-formation", () => {
    const { getByLabelText, getByText } = renderForm({
      contrat: "ctt"
    });
    const missionFormation = getByLabelText(/non/i);
    fireEvent.click(missionFormation);
    // blur is need to force validation in react-testing-lib
    fireEvent.blur(missionFormation);
    expect(getByText(/attention/i)).toBeTruthy();
  });
});
