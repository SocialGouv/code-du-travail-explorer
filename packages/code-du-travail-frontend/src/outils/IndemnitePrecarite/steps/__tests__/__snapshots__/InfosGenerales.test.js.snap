// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepInfosGenerales /> should render 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 2em;
  padding: 0;
  font-size: 1.8rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 2rem;
}

.c0 {
  display: block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  cursor: default;
}

.c1 {
  display: inline-block;
  margin-left: 1rem;
  color: #eb5757;
  font-weight: 700;
}

<div>
  <form>
    <p
      class="c0"
    >
      Quel est le type du contrat de travail ?
      <span
        aria-label="champs obligatoire"
        class="c1"
      >
        *
      </span>
    </p>
    <div
      class="c2"
    >
      <label
        class="c3"
      >
        <input
          name="contractType"
          type="radio"
          value="CDD"
        />
        <span>
          Contrat à durée déterminée (CDD)
        </span>
      </label>
      <label
        class="c3"
      >
        <input
          name="contractType"
          type="radio"
          value="CTT"
        />
        <span>
          Contrat de travail temporaire (Contrat d’intérim)
        </span>
      </label>
    </div>
    <button
      data-testid="nextBt"
    >
      suivant
    </button>
  </form>
</div>
`;

exports[`<StepInfosGenerales /> should render with CDD 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 2em;
  padding: 0;
  font-size: 1.8rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 2rem;
}

.c0 {
  display: block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  cursor: default;
}

.c5 {
  display: block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  cursor: pointer;
}

.c1 {
  display: inline-block;
  margin-left: 1rem;
  color: #eb5757;
  font-weight: 700;
}

.c6 {
  -webkit-flex: 1 1 70%;
  -ms-flex: 1 1 70%;
  flex: 1 1 70%;
  min-width: 400px;
  max-width: 100%;
  margin-top: 1rem;
  margin-right: 2rem;
}

.c4 {
  margin: 2rem 0;
}

<div>
  <form>
    <p
      class="c0"
    >
      Quel est le type du contrat de travail ?
      <span
        aria-label="champs obligatoire"
        class="c1"
      >
        *
      </span>
    </p>
    <div
      class="c2"
    >
      <label
        class="c3"
      >
        <input
          checked=""
          name="contractType"
          type="radio"
          value="CDD"
        />
        <span>
          Contrat à durée déterminée (CDD)
        </span>
      </label>
      <label
        class="c3"
      >
        <input
          name="contractType"
          type="radio"
          value="CTT"
        />
        <span>
          Contrat de travail temporaire (Contrat d’intérim)
        </span>
      </label>
    </div>
    <div
      class="c4"
    >
      <label
        class="c5"
        for="input-criteria.cddType"
      >
        Quelle est le type du cdd ?
      </label>
      <select
        class="c6"
        id="input-criteria.cddType"
        name="criteria.cddType"
      >
        <option
          disabled=""
          value=""
        >
          ...
        </option>
        <option
          value="CDD saisonnier"
        >
          CDD saisonnier
        </option>
        <option
          value="CDD d'usage"
        >
          CDD d'usage
        </option>
        <option
          value="CDD conclu avec un jeune (mineur ou majeur) pendant ses vacances scolaires ou universitaires"
        >
          CDD conclu avec un jeune (mineur ou majeur) pendant ses vacances scolaires ou universitaires
        </option>
        <option
          value="CCD dans le cadre d’un congé de mobilité"
        >
          CCD dans le cadre d’un congé de mobilité
        </option>
        <option
          value="Contrat unique d’insertion (CUI) ou Parcours emploi compétences (PEC)"
        >
          Contrat unique d’insertion (CUI) ou Parcours emploi compétences (PEC)
        </option>
        <option
          value="Contrat de professionnalisation ou Contrat d’apprentissage"
        >
          Contrat de professionnalisation ou Contrat d’apprentissage
        </option>
        <option
          value="Contrat pour lequel l’employeur s’est engagé à assurer un complément de formation professionnelle au salarié"
        >
          Contrat pour lequel l’employeur s’est engagé à assurer un complément de formation professionnelle au salarié
        </option>
        <option
          value="Autres"
        >
          Autres
        </option>
      </select>
    </div>
    <button
      data-testid="nextBt"
    >
      suivant
    </button>
  </form>
</div>
`;

exports[`<StepInfosGenerales /> should render with CDD and not handled ccn 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 2em;
  padding: 0;
  font-size: 1.8rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 2rem;
}

.c0 {
  display: block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  cursor: default;
}

.c5 {
  display: block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  cursor: pointer;
}

.c1 {
  display: inline-block;
  margin-left: 1rem;
  color: #eb5757;
  font-weight: 700;
}

.c6 {
  -webkit-flex: 1 1 70%;
  -ms-flex: 1 1 70%;
  flex: 1 1 70%;
  min-width: 400px;
  max-width: 100%;
  margin-top: 1rem;
  margin-right: 2rem;
}

.c4 {
  margin: 2rem 0;
}

<div>
  <form>
    <p
      class="c0"
    >
      Quel est le type du contrat de travail ?
      <span
        aria-label="champs obligatoire"
        class="c1"
      >
        *
      </span>
    </p>
    <div
      class="c2"
    >
      <label
        class="c3"
      >
        <input
          checked=""
          name="contractType"
          type="radio"
          value="CDD"
        />
        <span>
          Contrat à durée déterminée (CDD)
        </span>
      </label>
      <label
        class="c3"
      >
        <input
          name="contractType"
          type="radio"
          value="CTT"
        />
        <span>
          Contrat de travail temporaire (Contrat d’intérim)
        </span>
      </label>
    </div>
    <div
      class="c4"
    >
      <label
        class="c5"
        for="input-criteria.cddType"
      >
        Quelle est le type du cdd ?
      </label>
      <select
        class="c6"
        id="input-criteria.cddType"
        name="criteria.cddType"
      >
        <option
          disabled=""
          value=""
        >
          ...
        </option>
        <option
          value="CDD saisonnier"
        >
          CDD saisonnier
        </option>
        <option
          value="CDD d'usage"
        >
          CDD d'usage
        </option>
        <option
          value="CDD conclu avec un jeune (mineur ou majeur) pendant ses vacances scolaires ou universitaires"
        >
          CDD conclu avec un jeune (mineur ou majeur) pendant ses vacances scolaires ou universitaires
        </option>
        <option
          value="CCD dans le cadre d’un congé de mobilité"
        >
          CCD dans le cadre d’un congé de mobilité
        </option>
        <option
          value="Contrat unique d’insertion (CUI) ou Parcours emploi compétences (PEC)"
        >
          Contrat unique d’insertion (CUI) ou Parcours emploi compétences (PEC)
        </option>
        <option
          value="Contrat de professionnalisation ou Contrat d’apprentissage"
        >
          Contrat de professionnalisation ou Contrat d’apprentissage
        </option>
        <option
          value="Contrat pour lequel l’employeur s’est engagé à assurer un complément de formation professionnelle au salarié"
        >
          Contrat pour lequel l’employeur s’est engagé à assurer un complément de formation professionnelle au salarié
        </option>
        <option
          value="Autres"
        >
          Autres
        </option>
      </select>
    </div>
    <button
      data-testid="nextBt"
    >
      suivant
    </button>
  </form>
</div>
`;

exports[`<StepInfosGenerales /> should render with CTT 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 2em;
  padding: 0;
  font-size: 1.8rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 2rem;
}

.c0 {
  display: block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  cursor: default;
}

.c1 {
  display: inline-block;
  margin-left: 1rem;
  color: #eb5757;
  font-weight: 700;
}

<div>
  <form>
    <p
      class="c0"
    >
      Quel est le type du contrat de travail ?
      <span
        aria-label="champs obligatoire"
        class="c1"
      >
        *
      </span>
    </p>
    <div
      class="c2"
    >
      <label
        class="c3"
      >
        <input
          name="contractType"
          type="radio"
          value="CDD"
        />
        <span>
          Contrat à durée déterminée (CDD)
        </span>
      </label>
      <label
        class="c3"
      >
        <input
          checked=""
          name="contractType"
          type="radio"
          value="CTT"
        />
        <span>
          Contrat de travail temporaire (Contrat d’intérim)
        </span>
      </label>
    </div>
    <p
      class="c0"
    >
      S’agit-il d’un contrat de mission-formation ?
      <span
        aria-label="champs obligatoire"
        class="c1"
      >
        *
      </span>
    </p>
    <div
      class="c2"
      data-testid="cttFormation"
    >
      <label
        class="c3"
      >
        <input
          name="cttFormation"
          type="radio"
          value="true"
        />
        <span>
          Oui
        </span>
      </label>
      <label
        class="c3"
      >
        <input
          name="cttFormation"
          type="radio"
          value="false"
        />
        <span>
          Non
        </span>
      </label>
    </div>
    <p
      class="c0"
    >
      Le contrat d'intérim a-t-il été rompu avant la fin prévue pour une des raisons suivantes : la propre initiative du salarié, la faute grave du salarié, cas de force majeure&nsbp;?
      <span
        aria-label="champs obligatoire"
        class="c1"
      >
        *
      </span>
    </p>
    <div
      class="c2"
      data-testid="ruptureContratFauteGrave"
    >
      <label
        class="c3"
      >
        <input
          name="ruptureContratFauteGrave"
          type="radio"
          value="true"
        />
        <span>
          Oui
        </span>
      </label>
      <label
        class="c3"
      >
        <input
          name="ruptureContratFauteGrave"
          type="radio"
          value="false"
        />
        <span>
          Non
        </span>
      </label>
    </div>
    <p
      class="c0"
    >
      À la fin du contrat d'intérim, le salarié a-t-il été immédiatement embauché en CDI au sein de l'entreprise dans laquelle il effectuait sa mission ?
      <span
        aria-label="champs obligatoire"
        class="c1"
      >
        *
      </span>
    </p>
    <div
      class="c2"
      data-testid="propositionCDIFinContrat"
    >
      <label
        class="c3"
      >
        <input
          name="propositionCDIFinContrat"
          type="radio"
          value="true"
        />
        <span>
          Oui
        </span>
      </label>
      <label
        class="c3"
      >
        <input
          name="propositionCDIFinContrat"
          type="radio"
          value="false"
        />
        <span>
          Non
        </span>
      </label>
    </div>
    <p
      class="c0"
    >
      Le salarié a-t-il refusé la mise en œuvre de la souplesse prévue dans le contrat d’intérim ?
      <span
        aria-label="champs obligatoire"
        class="c1"
      >
        *
      </span>
    </p>
    <div
      class="c2"
      data-testid="refusSouplesse"
    >
      <label
        class="c3"
      >
        <input
          name="refusSouplesse"
          type="radio"
          value="true"
        />
        <span>
          Oui
        </span>
      </label>
      <label
        class="c3"
      >
        <input
          name="refusSouplesse"
          type="radio"
          value="false"
        />
        <span>
          Non
        </span>
      </label>
    </div>
    <button
      data-testid="nextBt"
    >
      suivant
    </button>
  </form>
</div>
`;
